### ISSUE TITLE
Not working for Genshin

### ISSUE THREAD

**Issue by alohadude3** [canaria3406/hoyolab-auto-sign#16]
The script seems to be no longer working for Genshin (Still works for Honkai and Star Rail).

**Comment by thanminh24** [comment]
I think that it might be some kind of anti-bot thing because when I use an alt to sign in manually there is no verification.
So the solution might be to try manual for a couple of days I guess

**Comment by TheTychoStar** [comment]
> 
I guess so. In this case the bot should return an error message to indicate the failure of signing in. However, it send an "OK" message to the discord server. This should be fixed I believe. 
Mihoyo added intensive anti-bot verification on the CN server which made auto sign-in nearly impossible. Hopefully, this won't happen to the international server.

**Comment by cd36th** [comment]
Might be a bit hard to fix the "OK" message, as this is what HoYoLAB is returning from the API request. HoYoLAB is telling us that the check-in worked when it didn't.

**Comment by Chillrend** [comment]
Can we maybe add user-agent, and x-http-referrer reader to the scripts and. The end-server are probably checking for user agents and referrer like mentioned here: https://github.com/canaria3406/hoyolab-auto-sign/issues/11#issuecomment-1588510392

I will try to add the code to the `header` below to my scripts, and will report the result here tomorrow:

```javascript
const header = {
    Cookie: token,
    'x-rpc-app_version': '2.34.1',
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36',
    'x-rpc-client_type': '4',
    'Referer': 'https://webstatic.mihoyo.com/',
    'Origin': 'https://webstatic.mihoyo.com'
  };
```

**Comment by cd36th** [comment]
this change to header worked for me

**Comment by Chillrend** [comment]
> this change to header worked for me

My account still requires captcha even after putting those header in the script.

The verification process probably done randomly or after certain days passed after checking-in without a user-agent/referrer, just a speculation though, cannot confirm anything...

Probably we want to push in the header changes to main anyway, just to make the script more like a human action to hoyoverse server logs.

**Comment by cd36th** [comment]
yeah i agree, do u wanna do the pull request? i dont wanna take credit for ur work lmao

**Comment by Chillrend** [comment]
I've opened one at https://github.com/canaria3406/hoyolab-auto-sign/pull/17 please have a look

**Comment by Chillrend** [comment]
Today, the script successfully checked in my genshin account without captcha verification.

Will report here again if the captcha came back.

**Comment by alohadude3** [comment]
A commit in a similar python project seems to include some challenge parameters in the header:
https://github.com/thesadru/genshin.py/commit/2d6f7d19485edb33c1529d50babd89dde64353c8
Though I'm not sure where you would get the actual solved challenge parameter from.

**Comment by djdoolky76** [comment]
seems to be working once again by doing nothing. although the captcha in the browser still showed up when you claim check in manually

**Comment by TheProJeckt** [comment]
It´s broken for me again :(
(Broke today)

(Telegramm)
Check-in completed for (me)
Genshin: Auto check-in failed due to CAPTCHA blocking. Please stop this script immediately and perform manual check-in for at least one week before attempting to use this script again.

After I manually collected it, it´s giving me the same message.

**Comment by Chillrend** [comment]
Mine also started breaking today.

I can see some repo (like one mentioned above) might able to successfully bypass the geetest captcha, but i would not want to do it. Pretty sure it's borderline against their ToS.

Will report here again if i find another solution.

**Comment by ChristopherKlay** [comment]
> Pretty sure it's borderline against their ToS.

The entire script already is.

**Comment by Deses** [comment]
I needed to change `const bannedCheck = hoyolabResponse.data?.gt_result?.is_risk;` to `const bannedCheck = hoyolabResponse.data.data?.gt_result?.is_risk;` (add one more `data`) because the first one returns undefined, there are two data nodes and we need to get the deepest one.

```
...
    [Symbol(kUniqueHeaders)]: null
  },
  data: {
    retcode: 0,
    message: 'OK',
    data: { code: 'ok', first_bind: false, gt_result: [Object] }
  }

```
